src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -2411,15 +2434,23 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 				arg0_type = MONO_TYPE_I8;
 				cast_class = mono_defaults.int64_class;
 			}
-			arg_class = create_class_instance ("System.Runtime.Intrinsics", m_class_get_name (klass), m_class_get_byval_arg (cast_class));
+
+			const char *klass_name = m_class_get_name (klass);
+
+			if (!strcmp (klass_name, "Vector4")) {
+				klass_name = "Vector128`1";
+			}
+			arg_class = create_class_instance ("System.Runtime.Intrinsics", klass_name, m_class_get_byval_arg (cast_class));
 		} else {
 			arg_class = mono_class_from_mono_type_internal (fsig->params [0]);
 		}
 
-		// FIXME: Add support for Vector64 on arm64 https://github.com/dotnet/runtime/issues/90402
-		int size = mono_class_value_size (arg_class, NULL);
-		if (size != 16)
+		size = mono_class_value_size (arg_class, NULL);
+
+		if (size != 16) {
```

Comment:
With the size check above, can this ever fail?

Reply:
It cannot, but its also a pre-existing condition that's tracking a separate piece of work since Vector64<T> acceleration is unique to Arm64.